synapseclient/models/protocols/file_protocol.py

```diff
@@ -0,0 +1,284 @@
+
+
+        Example: Using this function
+            File with the ID `syn123` at path `path/to/file.txt`:
+
+                file_instance = File(id="syn123", path="path/to/file.txt").store()
+
+            File at the path `path/to/file.txt` and a parent folder with the ID `syn456`:
+
+                file_instance = File(path="path/to/file.txt", parent_id="syn456").store()
+
+            File at the path `path/to/file.txt` and a parent folder with the ID `syn456`:
+
+                file_instance = File(path="path/to/file.txt").store(parent=Folder(id="syn456"))
+
+            Rename a file (Does not update the file on disk or the name of the downloaded file):
+
+                file_instance = File(id="syn123", download_file=False).get()
+                print(file_instance.name)  ## prints, e.g., "my_file.txt"
+                file_instance.change_metadata(name="my_new_name_file.txt")
+
+            Rename a file, and the name of the file as downloaded (Does not update the file on disk):
+
+                file_instance = File(id="syn123", download_file=False).get()
+                print(file_instance.name)  ## prints, e.g., "my_file.txt"
+                file_instance.change_metadata(name="my_new_name_file.txt", download_as="my_new_name_file.txt")
+
+        """
+        return self
+
+    def change_metadata(
+        self,
+        name: Optional[str] = None,
+        download_as: Optional[str] = None,
+        content_type: Optional[str] = None,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "File":
+        """
+        Change File Entity metadata for properties that are immutable after creation
+        through the store method.
+
+        Arguments:
+            name: Specify to change the filename of a file as seen on Synapse.
+            download_as: Specify filename to change the filename of a filehandle.
+            content_type: Specify content type to change the content type of a
+                filehandle.
+            synapse_client: If not passed in or None this will use the last client from
+                the `.login()` method.
+
+        Returns:
+            The file object.
+
+        Example: Using this function
+            Can be used to change the filename, the filename when the file is downloaded, or the file content-type without downloading:
+
+                file_entity = File(id="syn123", download_file=False).get()
+                print(os.path.basename(file_entity.path))  ## prints, e.g., "my_file.txt"
+                file_entity = file_entity.change_metadata(name="my_new_name_file.txt", download_as="my_new_downloadAs_name_file.txt", content_type="text/plain")
+                print(os.path.basename(file_entity.path))  ## prints, "my_new_downloadAs_name_file.txt"
+                print(file_entity.name) ## prints, "my_new_name_file.txt"
+
+        Raises:
+            ValueError: If the file does not have an ID to change metadata.
+        """
+        return self
+
+    def get(
```

Comment:
So I'm assuming that the functions here are all copied from the original models?